atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -633,3 +494,247 @@ def run(self):
+            }
+        if self.workflow.all_yum_repourls:
+            extra['image']['yum_repourls'] = self.workflow.all_yum_repourls
+
+        self.set_help(extra, worker_metadatas)
+        self.set_operators_metadata(extra, worker_metadatas)
+        self.set_remote_sources_metadata(extra)
+
+        self.set_go_metadata(extra)
+        self.set_group_manifest_info(extra, worker_metadatas)
+        extra['osbs_build']['kind'] = KOJI_KIND_IMAGE_BUILD
+        extra['osbs_build']['engine'] = self.workflow.builder.tasker.build_method
+        if has_operator_appregistry_manifest(self.workflow):
+            extra['osbs_build']['subtypes'].append(KOJI_SUBTYPE_OP_APPREGISTRY)
+        if has_operator_bundle_manifest(self.workflow):
+            extra['osbs_build']['subtypes'].append(KOJI_SUBTYPE_OP_BUNDLE)
+
+        return extra
+
+    def _update_build(self, build):
+        labels = Labels(df_parser(self.workflow.builder.df_path,
+                                  workflow=self.workflow).labels)
+        _, component = labels.get_name_and_value(Labels.LABEL_TYPE_COMPONENT)
+        _, version = labels.get_name_and_value(Labels.LABEL_TYPE_VERSION)
+        _, release = labels.get_name_and_value(Labels.LABEL_TYPE_RELEASE)
+        source = self.workflow.source
+        if not isinstance(source, GitSource):
+            raise RuntimeError('git source required')
+        build.update({
+            'name': component,
+            'version': version,
+            'release': release,
+            'source': "{0}#{1}".format(source.uri, source.commit_id),
+        })
+        return build
+
+    def get_server_dir(self):
+        """
+        Obtain koji_upload_dir value used for worker builds.
+        """
+        return get_koji_upload_dir(self.workflow)
+
+
+class KojiImportSourceContainerPlugin(KojiImportBase):
+
+    key = PLUGIN_KOJI_IMPORT_SOURCE_CONTAINER_PLUGIN_KEY  # type: ignore
+
+    def __init__(self, tasker, workflow,
+                 kojihub=None, url=None,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 blocksize=None,
+                 target=None, poll_interval=5):
+        super(KojiImportSourceContainerPlugin, self).__init__(tasker, workflow,
+                                                              kojihub, url,
+                                                              verify_ssl, use_auth,
+                                                              koji_ssl_certs, koji_proxy_user,
+                                                              koji_principal, koji_keytab,
+                                                              blocksize,
+                                                              target, poll_interval)
+        self.source_build = bool(self.workflow.build_result.oci_image_path)
```

Comment:
not needed anymore

Reply:
What?

Reply:
Ah. Somehow it wasn't connecting in my mind that this is (obviously) setting the `source_build` flag, which is what *was* used to switch through various the `if`s